### DISCUSSION TITLE
[ARM] No PPB emulation?

### DISCUSSION THREAD

**Discussion by Joshua-Schroijen** [unicorn-engine/unicorn#1956] [Q&A]
Hello everyone,

I noticed today that Unicorn doesn't intercept writes to the PPB (the 0xE000 0000 to 0xE0FF FFFF range in the M-profile system address map). My question is therefore if the Unicorn engine actually emulates the basic PPB functions like QEMU would have to do. And if it does, do I just still need to mem_map() that memory area?

Am I crazy or isn't this very commonly used by embedded software at start-up? I can't imagine I'm the first one to run into this limitation so I'm very curious to see how others have worked around this limitation and what plans for the future are regarding PPB emulation.  

Thank you in advance!  

Joshua

**Comment by wtdcode** [comment]
I think you should look at `uc_mmio_map`
